### PR TITLE
add more informations about compatibility between each components

### PR DESCRIPTION
**Pull request by geoffreyp**

### CODE REVIEW

File: moead_framework/core/selector/delta_selector.py [hunk 1]
```diff
@@ -5,10 +5,25 @@
...
+        Constructor of the mating pool selector
+
+        :param algorithm_instance: {:class:`~moead_framework.algorithm.abstract_moead.py.AbstractMoead`} instance of the algorithm
+        """
+        super().__init__(algorithm_instance)
+
+        try:
+            if not hasattr(self.algorithm, 'delta'):
+                raise
+        except Exception:
```
Comment (sjvrijn) [inline moead_framework/core/selector/delta_selector.py:23]: I think you can simplify the 
```python
try:
    if not hasattr(...):
        raise
except Exception:
    raise ValueError(msg)
```
blocks to the more concise
```python
if not hasattr(...):
    raise ValueError(msg)
```

File: moead_framework/core/selector/delta_selector.py [hunk 2]
```diff
@@ -5,10 +5,25 @@
...
+            msg = "The algorithm does not seem to be compatible with the component DeltaSelector. The parameter delta is required in the algorithm by the component DeltaSelector."
```
Comment (sjvrijn) [inline moead_framework/core/selector/delta_selector.py:24]: Error messages should be short and to the point.
Would summarize this one to: "Algorithm lacks required parameter 'delta' for component 'DeltaSelector'. "

File: moead_framework/core/selector/delta_selector.py [hunk 3]
```diff
@@ -5,10 +5,25 @@
...
+            raise ValueError(msg)
```
Comment (sjvrijn) [inline moead_framework/core/selector/delta_selector.py:25]: `AttributeError` seems like the better error here

File: moead_framework/core/sps_strategy/sps_dra.py [hunk 4]
```diff
@@ -21,6 +24,13 @@ def get_sub_problems(self):
 
         :return: {list<integer>} indexes of sub-problems
         """
+        try:
+            if not hasattr(self.algorithm, 'pi'):
+                raise
+        except Exception:
+            msg = "The algorithm does not seem to be compatible with the component SpsDra. The attribute pi is required in the algorithm to stock all utility values of each sub-problems."
```
Comment (sjvrijn) [inline moead_framework/core/sps_strategy/sps_dra.py:31]: "Algorithm lacks required parameter 'pi' for component 'SpsDra'."

File: moead_framework/core/sps_strategy/sps_dra.py [hunk 5]
```diff
@@ -21,6 +24,13 @@ def get_sub_problems(self):
...
+            raise ValueError(msg)
```
Comment (sjvrijn) [inline moead_framework/core/sps_strategy/sps_dra.py:32]: `AttributeError`

File: moead_framework/core/sps_strategy/sps_random_and_boundaries.py [hunk 6]
```diff
@@ -11,15 +11,25 @@ class SpsRandomAndBoundaries(SpsStrategy):
...
 
         lambda is represented here by the attribute self.algorithm.number_of_subproblem
 
         :return: {list<integer>} indexes of sub-problems
         """
+        try:
+            if not hasattr(self.algorithm, 'number_of_subproblem'):
+                raise
+        except Exception:
+            msg = "The algorithm does not seem to be compatible with the component SpsRandomAndBoundaries. The attribute number_of_subproblem is required in the algorithm to define the number of sub-problem to iterate."
```
Comment (sjvrijn) [inline moead_framework/core/sps_strategy/sps_random_and_boundaries.py:30]: Algorithm lacks required parameter 'number_of_subproblem' for component 'SpsRandomAndBoundaries'.

File: moead_framework/core/sps_strategy/sps_random_and_boundaries.py [hunk 7]
```diff
@@ -11,15 +11,25 @@ class SpsRandomAndBoundaries(SpsStrategy):
...
+            raise ValueError(msg)
```
Comment (sjvrijn) [inline moead_framework/core/sps_strategy/sps_random_and_boundaries.py:31]: `AttributeError`

File: moead_framework/core/genetic_operator/abstract_operator.py [hunk 8]
```diff
@@ -28,5 +28,6 @@ def number_of_solution_is_correct(self, n):
         :param n: {integer} number of solution required by the operator
         :return:
         """
-        if len(self.solutions) < n:
-            raise ValueError("The genetic operator needs more than "+str(n)+" solutions")
+        if len(self.solutions) != n:
+            msg = "The algorithm does not seem to be compatible with the current genetic operator. The current genetic oeperator needs {n} solutions as parents and it received {badn} solution(s)."
```
Comment (sjvrijn) [inline moead_framework/core/genetic_operator/abstract_operator.py:32]: "Selected genetic operator needs {n} solutions as parents, but received {badn} solution(s)"

Could be further specified using `self.__class__.__name__` if you'd want
```python
f"{self.__class__.__name__} needs {n} solutions as parents, but received {len(self.solutions)} solution(s)"
```

**Review by sjvrijn — CHANGES_REQUESTED** [review]

File: moead_framework/core/genetic_operator/abstract_operator.py (hunk 8 above)
Comment (geoffreyp) [reply moead_framework/core/genetic_operator/abstract_operator.py]: Fixed in 233c694

File: moead_framework/core/sps_strategy/sps_random_and_boundaries.py (hunk 7 above)
Comment (geoffreyp) [reply moead_framework/core/sps_strategy/sps_random_and_boundaries.py]: Fixed in 233c694

File: moead_framework/core/sps_strategy/sps_random_and_boundaries.py (hunk 6 above)
Comment (geoffreyp) [reply moead_framework/core/sps_strategy/sps_random_and_boundaries.py]: Fixed in 233c694

File: moead_framework/core/sps_strategy/sps_dra.py (hunk 5 above)
Comment (geoffreyp) [reply moead_framework/core/sps_strategy/sps_dra.py]: Fixed in 233c694

File: moead_framework/core/sps_strategy/sps_dra.py (hunk 4 above)
Comment (geoffreyp) [reply moead_framework/core/sps_strategy/sps_dra.py]: Fixed in 233c694

File: moead_framework/core/selector/delta_selector.py (hunk 3 above)
Comment (geoffreyp) [reply moead_framework/core/selector/delta_selector.py]: Fixed in 233c694

File: moead_framework/core/selector/delta_selector.py (hunk 2 above)
Comment (geoffreyp) [reply moead_framework/core/selector/delta_selector.py]: Fixed in 233c694

File: moead_framework/core/selector/delta_selector.py (hunk 1 above)
Comment (geoffreyp) [reply moead_framework/core/selector/delta_selector.py]: Fixed in 233c694

**Review by sjvrijn — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**